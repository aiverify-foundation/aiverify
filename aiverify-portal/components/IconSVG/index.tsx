import { ReactElement } from 'react';
import styles from './styles/icon.module.css';

import { AlertIcon } from './icons/alert-icon';
import { ArrowLeftIcon } from './icons/arrow-left-icon';
import { WideArrowLeftIcon } from './icons/wide-arrow-left-icon';
import { ArrowRightIcon } from './icons/arrow-right-icon';
import { WideArrowRightIcon } from './icons/wide-arrow-right-icon';
import { AsteriskIcon } from './icons/asterisk-icon';
import { BellIcon } from './icons/bell-icon';
import { BookIcon } from './icons/book-icon';
import { SolidBoxIcon } from './icons/solid-box-icon';
import { OutlineBoxIcon } from './icons/outline-box-icon';
import { BurgerMenuIcon } from './icons/burger-menu-icon';
import { ChatBubbleWideIcon } from './icons/chat-bubble-wide-icon';
import { ChatBubblesIcon } from './icons/chat-bubbles-icon';
import { CheckListIcon } from './icons/checklist-icon';
import { CircleArrowLeftIcon } from './icons/circle-arrow-left-icon';
import { CircleArrowRightIcon } from './icons/circle-arrow-right-icon';
import { CloseIcon } from './icons/close-x-icon';
import { DarkMoonIcon } from './icons/dark-moon-icon';
import { DocumentIcon } from './icons/document-icon';
import { FileIcon } from './icons/file-icon';
import { FolderForChatSessionsIcon } from './icons/folder-chat-icon';
import { FolderIcon } from './icons/folder-icon';
import { HistoryClockIcon } from './icons/history-clock-icon';
import { LayoutColumnsIcon } from './icons/layout-columns-icon';
import { LayoutWtfIcon } from './icons/layout-wtf-icon';
import { LightBulb } from './icons/light-bulb-icon';
import { LightSunIcon } from './icons/light-sun-icon';
import { LightningIcon } from './icons/lightning-icon';
import { ListIcon } from './icons/list-icon';
import { MaximizeIcon } from './icons/maximize-icon';
import { MinimizeIcon } from './icons/minimize-icon';
import { MoonshotAttackStrategyIcon } from './icons/ms-attack-icon';
import { MoonshotPromptTemplateIcon } from './icons/ms-prompt-template-icon';
import { PencilIcon } from './icons/pencil-icon';
import { PlusIcon } from './icons/plus-icon';
import { ResetIcon } from './icons/reset-icon';
import { RibbonIcon } from './icons/ribbon-icon';
import { RunCookbookIcon } from './icons/run-cookbook-icon';
import { SpacesuitIcon } from './icons/spacesuit-icon';
import { SquareIcon } from './icons/square-icon';
import { TableIcon } from './icons/table-icon';
import { TalkBubblesIcon } from './icons/talkbubbles-icon';
import { ToolsIcon } from './icons/tools-icon';
import { WarningIcon } from './icons/warning-icon';
import { WideArrowDownIcon } from './icons/wide-arrow-down';
import { WideArrowUpIcon } from './icons/wide-arrow-up';
import clsx from 'clsx';

enum IconName {
  Alert = 'Alert',
  ArrowLeft = 'ArrowLeft',
  ArrowRight = 'ArrowRight',
  Asterisk = 'Asterisk',
  Bell = 'Bell',
  Book = 'Book',
  BurgerMenu = 'BurgerMenu',
  ChatBubbleWide = 'ChatBubbleWide',
  ChatBubbles = 'ChatBubbles',
  CheckList = 'CheckList',
  CircleArrowLeft = 'CircleArrowLeft',
  CircleArrowRight = 'CircleArrowRight',
  Close = 'Close',
  DarkMoon = 'DarkMoon',
  Document = 'Document',
  File = 'File',
  Folder = 'Folder',
  FolderForChatSessions = 'FolderForChatSessions',
  HistoryClock = 'HistoryClock',
  LayoutColumns = 'LayoutColumns',
  LayoutWtf = 'LayoutWtf',
  LightBulb = 'LightBulb',
  LightSun = 'LightSun',
  Lightning = 'Lightning',
  List = 'List',
  Maximize = 'Maximize',
  Minimize = 'Minimize',
  MoonAttackStrategy = 'MoonAttackStrategy',
  MoonPromptTemplate = 'MoonPromptTemplate',
  OutlineBox = 'OutlineBox',
  Pencil = 'Pencil',
  Plus = 'Plus',
  Reset = 'Reset',
  Ribbon = 'Ribbon',
  RunCookbook = 'RunCookbook',
  SolidBox = 'SolidBox',
  Spacesuit = 'Spacesuit',
  Square = 'Square',
  Table = 'Table',
  TalkBubbles = 'TalkBubbles',
  Tools = 'Tools',
  Warning = 'Warning',
  WideArrowDown = 'WideArrowDown',
  WideArrowLeft = 'WideArrowLeft',
  WideArrowRight = 'WideArrowRight',
  WideArrowUp = 'WideArrowUp',
}

type BaseIconProps = {
  name: IconName;
  role?: string;
  ariaLabel?: string;
  size?: number;
  disabled?: boolean;
  style?: React.CSSProperties;
  onClick?: (e: React.MouseEvent<HTMLDivElement>) => void;
  onMouseDown?: (e: React.MouseEvent) => void;
};

type WithColor = BaseIconProps & {
  color: string;
  svgClassName?: never;
};

type WithClassName = BaseIconProps & {
  svgClassName: `stroke-${string} dark:stroke-${string}` | `fill-${string} dark:fill-${string}`;
  color?: never;
};

type IconProps = WithColor | WithClassName;

function Icon(props: IconProps) {
  const {
    name,
    role,
    ariaLabel,
    onClick,
    onMouseDown,
    disabled = false,
    size = 20,
    style,
    color = '#000000',
    svgClassName,
  } = props;

  let iconToRender: ReactElement | null = null;

  switch (name) {
    case IconName.Alert:
      iconToRender = (
        <AlertIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Folder:
      iconToRender = (
        <FolderIcon
          backColor="#2980b9"
          frontColor="#3498db"
          midColor="#bdc3c7"
          width={size}
          height={size}
        />
      );
      break;
    case IconName.FolderForChatSessions:
      iconToRender = (
        <FolderForChatSessionsIcon
          backColor="#2980b9"
          frontColor="#3498db"
          midColor="#bdc3c7"
          chatIconColor="#2980b9"
          width={size}
          height={size}
        />
      );
      break;
    case IconName.ChatBubbles:
      iconToRender = (
        <ChatBubblesIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.RunCookbook:
      iconToRender = (
        <RunCookbookIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.BurgerMenu:
      iconToRender = (
        <BurgerMenuIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.DarkMoon:
      iconToRender = (
        <DarkMoonIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.LightSun:
      iconToRender = (
        <LightSunIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Close:
      iconToRender = (
        <CloseIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.WideArrowLeft:
      iconToRender = (
        <WideArrowLeftIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.WideArrowRight:
      iconToRender = (
        <WideArrowRightIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.CircleArrowRight:
      iconToRender = (
        <CircleArrowRightIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.CircleArrowLeft:
      iconToRender = (
        <CircleArrowLeftIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Maximize:
      iconToRender = (
        <MaximizeIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Minimize:
      iconToRender = (
        <MinimizeIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.LayoutWtf:
      iconToRender = (
        <LayoutWtfIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.LayoutColumns:
      iconToRender = (
        <LayoutColumnsIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.ChatBubbleWide:
      iconToRender = (
        <ChatBubbleWideIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Reset:
      iconToRender = (
        <ResetIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.SolidBox:
      iconToRender = (
        <SolidBoxIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.OutlineBox:
      iconToRender = (
        <OutlineBoxIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Plus:
      iconToRender = <PlusIcon width={size} height={size} className={svgClassName} color={color} />;
      break;
    case IconName.Square:
      iconToRender = (
        <SquareIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.List:
      iconToRender = <ListIcon width={size} height={size} className={svgClassName} color={color} />;
      break;
    case IconName.File:
      iconToRender = <FileIcon width={size} height={size} className={svgClassName} color={color} />;
      break;
    case IconName.Book:
      iconToRender = <BookIcon width={size} height={size} className={svgClassName} color={color} />;
      break;
    case IconName.ArrowLeft:
      iconToRender = (
        <ArrowLeftIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.ArrowRight:
      iconToRender = (
        <ArrowRightIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Table:
      iconToRender = (
        <TableIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Bell:
      iconToRender = <BellIcon width={size} height={size} className={svgClassName} color={color} />;
      break;
    case IconName.Asterisk:
      iconToRender = (
        <AsteriskIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.CheckList:
      iconToRender = (
        <CheckListIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Spacesuit:
      iconToRender = (
        <SpacesuitIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.HistoryClock:
      iconToRender = (
        <HistoryClockIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Tools:
      iconToRender = (
        <ToolsIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Lightning:
      iconToRender = (
        <LightningIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.WideArrowDown:
      iconToRender = (
        <WideArrowDownIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.WideArrowUp:
      iconToRender = (
        <WideArrowUpIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.LightBulb:
      iconToRender = (
        <LightBulb width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Ribbon:
      iconToRender = (
        <RibbonIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.TalkBubbles:
      iconToRender = (
        <TalkBubblesIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Document:
      iconToRender = (
        <DocumentIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Warning:
      iconToRender = (
        <WarningIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.Pencil:
      iconToRender = (
        <PencilIcon width={size} height={size} className={svgClassName} color={color} />
      );
      break;
    case IconName.MoonPromptTemplate:
      iconToRender = (
        <MoonshotPromptTemplateIcon
          width={size}
          height={size}
          className={svgClassName}
          color={color}
        />
      );
      break;
    case IconName.MoonAttackStrategy:
      iconToRender = (
        <MoonshotAttackStrategyIcon
          width={size}
          height={size}
          className={svgClassName}
          color={color}
        />
      );
      break;
    default:
      iconToRender = null;
  }

  const cssClass = clsx(
    styles.icon_wrapper,
    onClick ? styles.pointer_effect : '',
    disabled ? styles.disabled : '',
  );

  return (
    <div
      role={role}
      aria-label={ariaLabel}
      className={cssClass}
      style={{
        width: size,
        height: size,
        ...style,
      }}
      onClick={disabled ? undefined : onClick}
      onMouseDown={disabled ? undefined : onMouseDown}
    >
      {iconToRender}
    </div>
  );
}

export { Icon, IconName };
