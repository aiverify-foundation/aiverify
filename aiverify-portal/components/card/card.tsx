import React, { useRef, useEffect } from 'react';
import VanillaTilt from 'vanilla-tilt';
import styles from './styles/card.module.css';
import clsx from 'clsx';

type CardProps = {
  size: 's' | 'm' | 'l';
  cardColor?: string;
  disableTiltEffect?: boolean;
  height?: number;
  style?: React.CSSProperties;
  className?: string;
  children?: React.ReactNode;
  onClick?: () => void;
};

function Card(props: CardProps) {
  const {
    size = 'm',
    cardColor,
    disableTiltEffect,
    height,
    style,
    className,
    children,
    onClick,
  } = props;

  const cardRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    if (cardRef.current && !disableTiltEffect) {
      VanillaTilt.init(cardRef.current, {
        max: 5,
        speed: 400,
        glare: true,
        'max-glare': 0.5,
      });
    }

    return () => {
      if (cardRef.current && !disableTiltEffect) {
        cardRef.current.vanillaTilt.destroy();
      }
    };
  }, [disableTiltEffect]);

  return (
    <figure
      ref={cardRef}
      className={clsx(
        styles.card,
        styles[`card_${size}`],
        'bg-gradient-to-b from-secondary-600 to-secondary-700 dark:bg-secondary-700',
        className,
      )}
      style={{ backgroundColor: cardColor, height, ...style }}
      onClick={onClick}
    >
      {children}
    </figure>
  );
}

export { Card };
