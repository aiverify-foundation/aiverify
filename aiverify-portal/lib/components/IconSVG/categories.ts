import { IconName } from '.';

export const lineIcons = [
  IconName.Alert,
  IconName.ArrowRight,
  IconName.ArrowLeft,
  IconName.Pencil,
  IconName.Asterisk,
  IconName.Book,
  IconName.BurgerMenu,
  IconName.ChatBubbleWide,
  IconName.ChatBubbles,
  IconName.CircleArrowRight,
  IconName.CircleArrowLeft,
  IconName.Close,
  IconName.Document,
  IconName.File,
  IconName.LayoutColumns,
  IconName.LayoutWtf,
  IconName.LightSun,
  IconName.List,
  IconName.Maximize,
  IconName.Minimize,
  IconName.OutlineBox,
  IconName.Pencil,
  IconName.Plus,
  IconName.Reset,
  IconName.RunCookbook,
  IconName.Square,
  IconName.Table,
  IconName.TalkBubbles,
  IconName.Warning,
  IconName.WideArrowUp,
  IconName.WideArrowDown,
  IconName.WideArrowLeft,
  IconName.WideArrowRight,
];

export const fillIcons = [
  IconName.Bell,
  IconName.CheckList,
  IconName.DarkMoon,
  IconName.Database,
  IconName.HistoryClock,
  IconName.LightBulb,
  IconName.Lightning,
  IconName.MoonPromptTemplate,
  IconName.Ribbon,
  IconName.SolidBox,
  IconName.Spacesuit,
  IconName.Tools,
  IconName.OpenedBook,
  IconName.OpenedEmptyBook,
  IconName.Plug,
];

export const colorIcons = [IconName.Folder, IconName.FolderForChatSessions];
