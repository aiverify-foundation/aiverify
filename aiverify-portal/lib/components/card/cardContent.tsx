import { PropsWithChildren } from 'react';
import styles from './styles/card.module.css';

type CardContentProps = {
  className?: string;
};

function CardContent({ children, className }: PropsWithChildren<CardContentProps>) {
  return <div className={`${styles.cardContent} ${className}`}>{children}</div>;
}

export { CardContent };
