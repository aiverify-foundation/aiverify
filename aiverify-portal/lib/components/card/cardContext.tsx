import React, { createContext, useContext } from 'react';

type CardContextType = {
  size: 'sm' | 'md' | 'lg';
};

const CardContext = createContext<CardContextType | undefined>(undefined);

export function CardProvider({
  value,
  children,
}: {
  value: CardContextType;
  children: React.ReactNode;
}) {
  return <CardContext.Provider value={value}>{children}</CardContext.Provider>;
}

export function useCardContext() {
  const context = useContext(CardContext);
  if (!context) {
    throw new Error('useCardContext must be used within a CardProvider');
  }
  return context;
}
