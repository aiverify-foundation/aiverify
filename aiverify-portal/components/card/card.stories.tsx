import type { Meta, StoryObj } from '@storybook/react';
import { Card } from './card';
import { Icon, IconName } from '../IconSVG';

const meta: Meta<typeof Card> = {
  title: 'Components/Card',
  component: Card,
  decorators: [
    (Story) => (
      <div
        style={{
          position: 'absolute',
          top: 0,
          width: '100%',
          height: '100vh',
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'center',
        }}
      >
        <Story />
      </div>
    ),
  ],
  argTypes: {
    size: {
      table: {
        disable: true,
      },
    },
    className: {
      table: {
        disable: true,
      },
    },
    cardColor: {
      table: {
        disable: true,
      },
    },
    height: {
      table: {
        disable: true,
      },
    },
    style: {
      table: {
        disable: true,
      },
    },
    children: {
      table: {
        disable: true,
      },
    },
    onClick: {
      table: {
        disable: true,
      },
    },
  },
  args: {
    disableTiltEffect: false,
  },
};

export default meta;
type Story = StoryObj<typeof Card>;

export const Default: Story = {
  args: {
    className: 'text-white flex flex-col gap-2 justify-between p-6',
    onClick: () => {
      alert('clicked');
    },
  },
  render: (args) => (
    <div className="flex gap-4">
      <Card {...args}>
        <Icon name={IconName.SolidBox} size={45} svgClassName="fill-white dark:fill-white" />
        <div className="flex flex-col gap-2">
          <h2 className="text-[1.2rem] font-bold">Card Title</h2>
          <figcaption>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut maximus purus sed velit
              porttitor, cursus auctor elit eleifend. Fusce ultricies ultrices sagittis.
            </p>
          </figcaption>
        </div>
      </Card>
      <Card {...args}>
        <Icon
          name={IconName.ChatBubbleWide}
          size={45}
          svgClassName="stroke-white dark:stroke-white"
        />
        <div className="flex flex-col gap-2">
          <h2 className="text-[1.2rem] font-bold">Card Title</h2>
          <figcaption>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut maximus purus sed velit
              porttitor, cursus auctor elit eleifend. Fusce ultricies ultrices sagittis.
            </p>
          </figcaption>
        </div>
      </Card>
      <Card {...args}>
        <Icon name={IconName.LightBulb} size={45} svgClassName="fill-white dark:fill-white" />
        <div className="flex flex-col gap-2">
          <h2 className="text-[1.2rem] font-bold">Card Title</h2>
          <figcaption>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut maximus purus sed velit
              porttitor, cursus auctor elit eleifend. Fusce ultricies ultrices sagittis.
            </p>
          </figcaption>
        </div>
      </Card>
    </div>
  ),
};
