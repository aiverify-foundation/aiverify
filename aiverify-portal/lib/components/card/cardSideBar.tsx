import { useCardContext } from './cardContext';
import styles from './styles/card.module.css';

type CardSideBarProps = {
  children: React.ReactNode;
  percentageWidth?: `${number}%`;
  className?: string;
};

function CardSideBar({
  children,
  percentageWidth,
  className,
}: CardSideBarProps) {
  const { size } = useCardContext();
  return (
    <div
      style={{ width: percentageWidth }}
      className={`${styles.cardSideBar} ${styles[`cardSideBar_${size}`]} ${className}`}>
      {children}
    </div>
  );
}

export { CardSideBar };
