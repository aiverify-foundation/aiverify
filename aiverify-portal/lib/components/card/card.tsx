'use client';

import React, { useRef, useEffect } from 'react';
import VanillaTilt from 'vanilla-tilt';
import { CardContent } from './cardContent';
import { CardProvider } from './cardContext';
import { CardSideBar } from './cardSideBar';
import styles from './styles/card.module.css';

type CommonCardProps = {
  size: 'sm' | 'md' | 'lg';
  cardColor?: string;
  width?: number;
  height?: number;
  style?: React.CSSProperties;
  className?: string;
  onClick?: () => void;
  children: React.ReactNode;
};

type BaseTiltProps = {
  tiltRotation?: number;
  tiltSpeed?: number;
};

type TiltPropsWithoutGlare = BaseTiltProps & {
  enableTiltGlare?: false;
  tiltMaxGlare?: never;
};

type TiltPropsWithGlare = BaseTiltProps & {
  enableTiltGlare: true;
  tiltMaxGlare: number;
};

type TiltProps = TiltPropsWithoutGlare | TiltPropsWithGlare;

type CardWithoutTilt = CommonCardProps & {
  enableTiltEffect?: false | undefined;
  tiltRotation?: never;
  tiltSpeed?: never;
  enableTiltGlare?: never;
  tiltMaxGlare?: never;
};

type CardWithTilt = CommonCardProps &
  TiltProps & {
    enableTiltEffect: true;
  };

type CardProps = CardWithoutTilt | CardWithTilt;

function Card(props: CardProps) {
  const {
    size = 'md',
    cardColor,
    enableTiltEffect,
    height,
    width,
    style,
    className,
    children,
    onClick,
    tiltRotation,
    tiltSpeed,
    enableTiltGlare,
    tiltMaxGlare,
  } = props;
  const cardRef = useRef<
    HTMLDivElement & { vanillaTilt: { destroy: () => void } }
  >(null);

  useEffect(() => {
    const tiltedElem = cardRef.current;

    if (tiltedElem && enableTiltEffect) {
      VanillaTilt.init(tiltedElem, {
        max: tiltRotation,
        speed: tiltSpeed,
        glare: enableTiltGlare,
        'max-glare': tiltMaxGlare,
      });
    }

    return () => {
      if (tiltedElem && enableTiltEffect) {
        tiltedElem.vanillaTilt.destroy();
      }
    };
  }, [
    enableTiltEffect,
    tiltRotation,
    tiltSpeed,
    enableTiltGlare,
    tiltMaxGlare,
  ]);

  return (
    <CardProvider value={{ size }}>
      <div
        ref={cardRef}
        className={`${styles.card} ${styles[`card_${size}`]} ${className}`}
        style={{
          height,
          width,
          ...(cardColor
            ? { backgroundColor: cardColor }
            : {
                backgroundImage:
                  'linear-gradient(to bottom, var(--color-primary-900), var(--color-primary-700))',
              }),
          ...style,
        }}
        onClick={onClick}>
        <div className={styles.cardFlexbox}>{children}</div>
      </div>
    </CardProvider>
  );
}

Card.SideBar = CardSideBar;
Card.Content = CardContent;
export { Card };
