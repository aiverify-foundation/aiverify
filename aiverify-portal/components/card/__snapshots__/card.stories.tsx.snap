// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Card Default smoke-test 1`] = `
<div style="position: relative; width: 100%; height: 300px; display: flex; align-items: center; justify-content: center;">
  <div style="position: absolute; top: 0px; width: 100%; height: 100vh; display: flex; align-items: center; justify-content: center;">
    <div class="flex gap-4">
      <figure class="_card_8npgw_1 _card_m_8npgw_37 bg-gradient-to-b from-secondary-600 to-secondary-700 dark:bg-secondary-700 text-white flex flex-col gap-2 justify-between p-6"
              style="will-change: transform; transition: 400ms cubic-bezier(0.03, 0.98, 0.52, 0.99); transform: perspective(1000px) rotateX(0deg) rotateY(0deg) scale3d(1, 1, 1);"
      >
        <div class="_icon_wrapper_1j07s_1"
             style="width: 45px; height: 45px;"
        >
          <svg fill="#000000"
               stroke="none"
               height="45"
               viewbox="0 0 15 15"
               width="45"
               xmlns="http://www.w3.org/2000/svg"
               class="fill-white dark:fill-white"
          >
            <path d="M7.30298 0.0404275C7.42875 -0.0134758 7.57112 -0.0134758 7.6969 0.0404275L14.5 2.95605L7.50002 5.95604L0.499949 2.95601L7.30298 0.0404275Z">
            </path>
            <path d="M-6.10352e-05 3.82969V11.5C-6.10352e-05 11.7 0.119137 11.8808 0.302979 11.9596L6.99994 14.8297V6.82969L-6.10352e-05 3.82969Z">
            </path>
            <path d="M7.99994 6.82976L14.9999 3.82976V11.5C14.9999 11.7 14.8807 11.8808 14.6969 11.9596L7.99994 14.8297V6.82976Z">
            </path>
          </svg>
        </div>
        <div class="flex flex-col gap-2">
          <h2 class="text-[1.2rem] font-bold">
            Card Title
          </h2>
          <figcaption>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut maximus purus sed velit porttitor, cursus auctor elit eleifend. Fusce ultricies ultrices sagittis.
            </p>
          </figcaption>
        </div>
        <div class="js-tilt-glare"
             style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; overflow: hidden; pointer-events: none; border-radius: inherit;"
        >
          <div class="js-tilt-glare-inner"
               style="position: absolute; top: 50%; left: 50%; pointer-events: none; background-image: linear-gradient(0deg, rgba(255, 255, 255, 0) 0%, rgb(255, 255, 255) 100%); transform: rotate(180deg) translate(-50%, -50%); transform-origin: 0% 0%; opacity: 0; width: 600px; height: 600px; transition: opacity 400ms cubic-bezier(0.03, 0.98, 0.52, 0.99);"
          >
          </div>
        </div>
      </figure>
      <figure class="_card_8npgw_1 _card_m_8npgw_37 bg-gradient-to-b from-secondary-600 to-secondary-700 dark:bg-secondary-700 text-white flex flex-col gap-2 justify-between p-6"
              style="will-change: transform; transition: 400ms cubic-bezier(0.03, 0.98, 0.52, 0.99); transform: perspective(1000px) rotateX(0deg) rotateY(0deg) scale3d(1, 1, 1);"
      >
        <div class="_icon_wrapper_1j07s_1"
             style="width: 45px; height: 45px;"
        >
          <svg xmlns="http://www.w3.org/2000/svg"
               fill="none"
               viewbox="0 0 24 24"
               width="45"
               height="45"
               stroke-width="1.5"
               stroke="#000000"
               class="stroke-white dark:stroke-white"
          >
            <path stroke-linecap="round"
                  stroke-linejoin="round"
                  d="M8.625 9.75a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Zm0 0H8.25m4.125 0a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Zm0 0H12m4.125 0a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Zm0 0h-.375m-13.5 3.01c0 1.6 1.123 2.994 2.707 3.227 1.087.16 2.185.283 3.293.369V21l4.184-4.183a1.14 1.14 0 0 1 .778-.332 48.294 48.294 0 0 0 5.83-.498c1.585-.233 2.708-1.626 2.708-3.228V6.741c0-1.602-1.123-2.995-2.707-3.228A48.394 48.394 0 0 0 12 3c-2.392 0-4.744.175-7.043.513C3.373 3.746 2.25 5.14 2.25 6.741v6.018Z"
            >
            </path>
          </svg>
        </div>
        <div class="flex flex-col gap-2">
          <h2 class="text-[1.2rem] font-bold">
            Card Title
          </h2>
          <figcaption>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut maximus purus sed velit porttitor, cursus auctor elit eleifend. Fusce ultricies ultrices sagittis.
            </p>
          </figcaption>
        </div>
        <div class="js-tilt-glare"
             style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; overflow: hidden; pointer-events: none; border-radius: inherit;"
        >
          <div class="js-tilt-glare-inner"
               style="position: absolute; top: 50%; left: 50%; pointer-events: none; background-image: linear-gradient(0deg, rgba(255, 255, 255, 0) 0%, rgb(255, 255, 255) 100%); transform: rotate(180deg) translate(-50%, -50%); transform-origin: 0% 0%; opacity: 0; width: 600px; height: 600px; transition: opacity 400ms cubic-bezier(0.03, 0.98, 0.52, 0.99);"
          >
          </div>
        </div>
      </figure>
      <figure class="_card_8npgw_1 _card_m_8npgw_37 bg-gradient-to-b from-secondary-600 to-secondary-700 dark:bg-secondary-700 text-white flex flex-col gap-2 justify-between p-6"
              style="will-change: transform; transition: 400ms cubic-bezier(0.03, 0.98, 0.52, 0.99); transform: perspective(1000px) rotateX(0deg) rotateY(0deg) scale3d(1, 1, 1);"
      >
        <div class="_icon_wrapper_1j07s_1"
             style="width: 45px; height: 45px;"
        >
          <svg width="45"
               height="45"
               viewbox="0 0 23 23"
               fill="#000000"
               class="fill-white dark:fill-white"
               xmlns="http://www.w3.org/2000/svg"
          >
            <path d="M14.9656 19.2539H8.03471C8.02797 19.3423 8.02344 19.4313 8.02344 19.5213C8.02344 21.4384 9.58313 22.9981 11.5002 22.9981C13.4172 22.9981 14.9769 21.4384 14.9769 19.5213C14.9769 19.4313 14.9724 19.3423 14.9656 19.2539Z">
            </path>
            <path d="M17.0001 2.26725C15.5263 0.803787 13.575 0 11.4995 0C11.4807 0 11.4616 8.98438e-05 11.4428 0.000179688C9.40903 0.0145547 7.48561 0.813131 6.02682 2.24888C4.56771 3.68499 3.73899 5.59547 3.69344 7.62868C3.64555 9.7631 4.44422 11.7743 5.94219 13.292C6.75397 14.1144 7.21958 15.2154 7.2199 16.3129V17.6512H15.7781V16.3129C15.7783 15.209 16.2361 14.1159 17.034 13.3139C18.4995 11.8408 19.3065 9.88528 19.3065 7.8076C19.3065 5.71168 18.4874 3.74406 17.0001 2.26725ZM10.0069 9.43503L8.8723 10.5697L6.6201 8.31742L8.87235 6.06522L10.007 7.19985L8.88942 8.31737L10.0069 9.43503ZM11.6488 11.5H10.0091L11.3491 5.13488H12.9889L11.6488 11.5ZM14.1256 10.5696L12.991 9.43494L14.1085 8.31737L12.991 7.1999L14.1256 6.06526L16.3779 8.31742L14.1256 10.5696Z">
            </path>
          </svg>
        </div>
        <div class="flex flex-col gap-2">
          <h2 class="text-[1.2rem] font-bold">
            Card Title
          </h2>
          <figcaption>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut maximus purus sed velit porttitor, cursus auctor elit eleifend. Fusce ultricies ultrices sagittis.
            </p>
          </figcaption>
        </div>
        <div class="js-tilt-glare"
             style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; overflow: hidden; pointer-events: none; border-radius: inherit;"
        >
          <div class="js-tilt-glare-inner"
               style="position: absolute; top: 50%; left: 50%; pointer-events: none; background-image: linear-gradient(0deg, rgba(255, 255, 255, 0) 0%, rgb(255, 255, 255) 100%); transform: rotate(180deg) translate(-50%, -50%); transform-origin: 0% 0%; opacity: 0; width: 600px; height: 600px; transition: opacity 400ms cubic-bezier(0.03, 0.98, 0.52, 0.99);"
          >
          </div>
        </div>
      </figure>
    </div>
  </div>
</div>
`;
